### PR TITLE
Create more cypress tests and utilities

### PR DESCRIPTION
**Pull request by t3chguy**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: cypress/plugins/synapsedocker/index.ts [hunk 1]
```diff
@@ -110,26 +129,27 @@ async function synapseStart(template: string): Promise<SynapseInstance> {
...
-    const port = await new Promise<number>((resolve, reject) => {
-        childProcess.execFile('docker', [
-            "port", synapseId, "8008",
+    synapses.set(synapseId, { synapseId, ...synCfg });
+
+    console.log(`Started synapse with id ${synapseId} on port ${synCfg.port}.`);
+
+    // Await Synapse healthcheck
+    await new Promise<void>((resolve, reject) => {
+        childProcess.execFile("docker", [
```
[the remark this answers is not in the document]
Comment (dbkr) [reply cypress/plugins/synapsedocker/index.ts:138]: Oh, I guess you get the retries / timeout for free, and you know there's a curl in the docker container. If I've guessed correctly please comment. :)

**Review by jryans — APPROVED** [review]
Seems like a reasonable step forward! 😄 Perhaps we'll later find we want to split these steps up a bit more, but seems good for now.

File: cypress/support/login.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+             * @param displayName the displayName to give the test user
+             */
+            initTestUser(synapse: SynapseInstance, displayName: string): Chainable<IUserCredentials>;
+        }
+    }
+}
+
+Cypress.Commands.add("initTestUser", (synapse: SynapseInstance, displayName: string): Chainable<IUserCredentials> => {
+    const username = Cypress._.uniqueId("userId_");
+    const password = Cypress._.uniqueId("password_");
```
Comment (dbkr) [inline cypress/support/login.ts:46]: This is fun - is this a cypress internal or something and what's it doing?

**Review by dbkr — COMMENTED** [review]
Generally looks great - could you also update docs/cypress.md with the state of play?

File: cypress/plugins/synapsedocker/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (t3chguy) [reply cypress/plugins/synapsedocker/index.ts:138]: Yes correct, we can't use `cy.request` as it doesn't retry on total connection failure
We could use `http` but that's horribly unusable, don't want to have to bring in another dep like `fetch` or `axios` where it isn't needed

File: cypress/support/login.ts (hunk 2 above)
Comment (t3chguy) [reply cypress/support/login.ts:46]: Its a re-exported lodash

**Review by dbkr — APPROVED** [review]
Thanks!

**Pull request merged**